pkg/queue/health/probe_test.go

```diff
@@ -126,14 +126,13 @@ func TestHTTPSchemeProbeSuccess(t *testing.T) {
 }
 
 func TestHTTPProbeTimeoutFailure(t *testing.T) {
-	timeout := 10 * time.Millisecond
 	server := newTestServer(t, func(w http.ResponseWriter, r *http.Request) {
-		time.Sleep(timeout * 5)
+		time.Sleep(50 * time.Millisecond)
```

Comment:
wdyt about something like this:

~~~~
	ch := make(chan struct{})
	defer close(ch)

	server := newTestServer(t, func(w http.ResponseWriter, r *http.Request) {
		select {
		case <-time.After(1 * time.Second):
		case <-ch:
		}

		w.WriteHeader(http.StatusOK)
	})
~~~~

that way we can have a long enough timeout to avoid any flakes, but also the test will end straight away once the probe fails.

Reply:
Hmm, good  point, I'll experiment a bit!

Reply:
Pushed, using the request context is even a bit prettier :sweat_smile:

Reply:
ooh yay, yep, even better!